code/modules/research/xenobiology/crossbreeding/regenerative.dm

```diff
@@ -23,9 +23,15 @@ Regenerative extracts:
 		to_chat(user, "<span class='warning'>[src] will not work on the dead!</span>")
 		return
 	if(H != user)
+		if(!do_mob(user, H, 50))
+			to_chat(user, "<span class='notice'>You need to hold still to apply [src]!")
+			return
 		user.visible_message("<span class='notice'>[user] crushes the [src] over [H], the milky goo quickly regenerating all of [H.p_their()] injuries!</span>",
 			"<span class='notice'>You squeeze the [src], and it bursts over [H], the milky goo regenerating [H.p_their()] injuries.</span>")
 	else
+		if(!do_mob(user, user, 50))
+			to_chat(user, "<span class='notice'>You need to hold still to apply [src]!")
+			return
```

Comment:
This could be done before the H != user check, as H will always be the variable being checked.
In case 1,  you are checking H, in case 2 you are checking user (which will be equal to H), meaning it is impossible to check anything other than H